livekit-plugins/livekit-plugins-playai/livekit/plugins/playai/tts.py

```diff
@@ -0,0 +1,313 @@
+    voice_guidance: float | None
+    style_guidance: float | None
+    repetition_penalty: float | None
+
+
+class TTS(tts.TTS):
+    def __init__(
+        self,
+        *,
+        api_key: str | None = None,
+        user_id: str | None = None,
+        voice: str = "s3://voice-cloning-zero-shot/775ae416-49bb-4fb6-bd45-740f205d20a1/jennifersaad/manifest.json",
+        language: str = "english",
+        sample_rate: int = 24000,
+        speed: float = 1.0,
+        model: TTSModel | str = "Play3.0-mini",
+        temperature: float | None = None,
+        top_p: float | None = None,
+        text_guidance: float | None = None,
+        voice_guidance: float | None = None,
+        style_guidance: float | None = None,
+        repetition_penalty: float | None = None,
+    ) -> None:
+        """
+        Initialize the PlayAI TTS engine.
+
+        Args:
+            api_key (str): The PlayAI API key. Can be set via environment variable PLAYAI_API_KEY.
+            user_id (str): The PlayAI user ID. Can be set via environment variable PLAYAI_USER_ID.
+            voice (str): A URL pointing to a Play voice manifest file. (e.g. "s3://voice-cloning-zero-shot/775ae416-49bb-4fb6-bd45-740f205d20a1/jennifersaad/manifest.json").
+            language (str): The language of the text. Default is 'english'.
+            sample_rate (int): The sample rate in Hz. Options are 8000, 16000, 24000, 44100, 48000.
+            speed (float): The speed of the audio. Default is 1.0.
+            model (str): The voice engine to use. Default is "Play3.0-mini".
+            > The following options are inference-time hyperparameters of the text-to-speech model; if unset, the model will use default values chosen by PlayAI.
+            temperature (float): The temperature of the model.
+            top_p (float): The top-p value of the model.
+            text_guidance (float): The text guidance of the model.
+            voice_guidance (float): The voice guidance of the model.
+            style_guidance (float): (Play3.0-mini-http and Play3.0-mini-ws only) The style guidance of the model.
+            repetition_penalty (float): The repetition penalty of the model.
+        """
+        super().__init__(
+            capabilities=tts.TTSCapabilities(
+                streaming=True,
+            ),
+            sample_rate=sample_rate,
+            num_channels=1,
+        )
+
+        api_key = api_key or os.environ.get("PLAYAI_API_KEY")
+        user_id = user_id or os.environ.get("PLAYAI_USER_ID")
+
+        if not api_key or not user_id:
+            raise ValueError(
+                "PlayAI API key and user ID are required, either as arguments or set PLAYAI_API_KEY and PLAYAI_USER_ID environment variables"
+            )
+
+        self._client = PlayHTAsyncClient(
+            user_id=user_id,
+            api_key=api_key,
+        )
+        # suppress verbose websocket logs
+        logging.getLogger("websockets.client").setLevel(logging.INFO)
```

Comment:
This should be done inside the log.py in the cli package

Reply:
yeah good point about doing it here.. this might not be an issue anymore, the team suggested using their websockets directly, since that supports play.ai keys. I'll update this later on